components/convert_channel_modal/index.ts

```diff
@@ -1,27 +1,29 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators} from 'redux';
+import React from 'react';
+import {AnyAction, bindActionCreators, Dispatch} from 'redux';
 import {connect} from 'react-redux';
 
 import {convertChannelToPrivate} from 'mattermost-redux/actions/channels';
 
 import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {GlobalState} from 'mattermost-redux/types/store';
 
-import ConvertChannelModal from './convert_channel_modal.jsx';
+import ConvertChannelModal from './convert_channel_modal';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     return {
         currentTeamDetails: getCurrentTeam(state),
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
```

Comment:
```suggestion
function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
```
Use `GenericAction` instead of `AnyAction`